x-pack/platform/plugins/shared/streams/server/routes/streams/ingest/route.ts

```diff
@@ -75,6 +75,13 @@ const upsertIngestRoute = createServerRoute({
       request,
     });
 
+    if (
+      !(await streamsClient.isStreamsEnabled()) &&
+      isWiredStreamDefinition({ name: params.path.name, ...params.body })
```

Comment:
Flipping these conditions around will save us an async call to `streamsClient.isStreamsEnabled` in the happy path